src/maximum_cardinality_matching.jl

```diff
@@ -0,0 +1,118 @@
+"""
+maximum_cardinality_matching(g::Graph)
+
+Given a graph `g` returns a maximum cardinality matching.
+This is the same as running `maximum_weight_matching` with default weights (`1`)
+but is faster without needing a JuMP solver.
+
+Returns MatchingResult containing:
+  - the maximum cardinality that can be achieved
+  - a list of each vertex's match (or -1 for unmatched vertices)
+"""
+function maximum_cardinality_matching(g::AbstractGraph{U}) where U<:Integer
```

Comment:
```suggestion
function maximum_cardinality_matching(g::AbstractGraph{U}) where {U<:Integer}
```

Nothing major, just consistency